src/main/java/uk/gov/justice/hmpps/offenderevents/services/MergeRecordDiscriminator.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.justice.hmpps.offenderevents.services;
+
+import com.microsoft.applicationinsights.TelemetryClient;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Component
+@Slf4j
+public class MergeRecordDiscriminator {
+    private final PrisonApiService prisonApiService;
+
+    private final TelemetryClient telemetryClient;
+
+    public MergeRecordDiscriminator(PrisonApiService prisonApiService,
+                                    TelemetryClient telemetryClient) {
+        this.prisonApiService = prisonApiService;
+        this.telemetryClient = telemetryClient;
+    }
+
+    public List<MergeOutcome> identifyMergedPrisoner(Long bookingId) {
+
+        return prisonApiService.getPrisonerNumberForBookingId(bookingId)
+            .map(prisonerNumber -> {
+                log.debug("Check for merges for booking ID {}, prisoner Number {}", bookingId, prisonerNumber);
+                final var mergedOffenders = prisonApiService.getIdentifiersByBookingId(bookingId);
+
+                return mergedOffenders.stream()
+                    .peek(mergedNumber -> {
```

Comment:
Is there a reason you have chosen peek here instead of for each? 

Although it may work, it might be best to avoid using it as it is intended ['mainly for debugging'](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#peek-java.util.function.Consumer-)